app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -80,13 +81,17 @@ class RecentSyncedTabFeature(
         lastSyncedTab = syncedTab
     }
 
-    // UI will either not be displayed if not authenticated (DefaultPresenter.start),
-    // or the display state will be tied directly to the success and error cases.
-    override fun stopLoading() = Unit
+    // This is called in success cases as well, but the state should only change if there isn't a
+    // tab displayed. The store's state isn't updated in time to rely on it for this condition,
+    // so local state is used instead.
+    override fun stopLoading() {
+        if (lastSyncedTab == null) {
```

Comment:
When the feature is detached and then destroyed, the `lastSyncedTab` will always be null. So the `stopLoading`  check would also always be true, right? What are the other cases this check will help with?

Reply:
I'm not sure I'm following. Are you suggesting that `stopLoading` is only called when the feature is destroyed? It is also called at end of every call to `DefaultController::refreshSyncedTabs`. This is mostly meant as a sanity check - normally we'll hit either the success case and `lastSyncedTab` will not be null or a failure case and the error will cause a `NONE` state dispatch.

Reply:
Ack, sorry - I misread this the other day. LGTM.